src/app/shared/abstracts/base-product/base-product.component.ts

```diff
@@ -66,4 +39,36 @@ export abstract class BaseProductComponent {
     }
 
     constructor(protected fb: FormBuilder) {}
+
+    ngOnInit(): void {
+        this.optionsForm = this.fb.group({
+            ...this.formConfig,
+            quantity: [1, Validators.required],
+        });
+    }
+
+    private get formConfig() {
+        return this.product
+            .attributes
+            .reduce(
+                (obj, attr) => {
+                    return {
+                        [attr.name]: [attr.options[0].optionName, Validators.required],
+                        ...obj
+                    }
+                }, {}
+            );
+    }
+
+    getAttributeOptions(attribute: string): string[] {
+        const attr = this.product
```

Comment:
here, we should loop through product attributes in the html file, that way we would only access one given attribute at a time, which would prevent having to use .find.

Also, it would greatly simplify because we can simply pass an options array (instead of the whole attribute object) and map through names